service/provider/src/main/java/gov/va/vro/service/provider/camel/BgsApiClientRoutes.java

```diff
@@ -0,0 +1,162 @@
+package gov.va.vro.service.provider.camel;
+
+import gov.va.vro.camel.RabbitMqCamelUtils;
+import gov.va.vro.camel.ToRabbitMqRouteHelper;
+import gov.va.vro.camel.processor.FunctionProcessor;
+import gov.va.vro.camel.processor.RequestAndMerge;
+import gov.va.vro.model.bgs.BgsApiClientRequest;
+import gov.va.vro.model.bgs.BgsApiClientResponse;
+import gov.va.vro.service.provider.MasConfig;
+import gov.va.vro.service.provider.bgs.service.BgsApiClient;
+import gov.va.vro.service.provider.bgs.service.BgsNotesCamelBody;
+import lombok.RequiredArgsConstructor;
+import lombok.extern.slf4j.Slf4j;
+import org.apache.camel.ExchangePattern;
+import org.apache.camel.ProducerTemplate;
+import org.apache.camel.builder.RouteBuilder;
+import org.apache.commons.lang3.StringUtils;
+import org.springframework.stereotype.Component;
+
+import java.util.concurrent.atomic.AtomicInteger;
+
+@Slf4j
+@Component
+@RequiredArgsConstructor
+public class BgsApiClientRoutes extends RouteBuilder {
+
+  public static final String ADD_BGS_NOTES = "direct:addBgsNotes-start";
+  static final String BGSCLIENT_ADDNOTES = "direct:toRabbit-bgsClient-addNotes";
+  static final String ADD_NOTES_RETRIES = "direct:addBgsNotesWithRetries";
+
+  private final ProducerTemplate producerTemplate;
+
+  private final BgsApiClient bgsApiClient;
+
+  int RETRY_LIMIT = 5;
+
+  @Override
+  public void configure() throws Exception {
+    var routeId = "add-bgs-notes-route";
+    configureAddNotesRoute(routeId);
+    configureRouteToBgsApiClientMicroservice();
+    configureRouteToSlackNotification();
+
+    configureMockBgsApiMicroservice();
+  }
+
+  private void configureAddNotesRoute(String routeId) {
+    from(ADD_BGS_NOTES)
+        .routeId(routeId)
+        .setExchangePattern(ExchangePattern.InOut)
+        // expecting body MasProcessingObject.class
+        .log("input: ${exchange.pattern}: h=${headers}: p=${exchange.properties}: ${body.class}")
+        .bean(bgsApiClient, "buildRequest")
+        .loopDoWhile(simple("${body.pendingRequests.size} > 0"))
+        .log("loop: ${body.pendingRequests.size} pendingRequests")
+        .to(ADD_NOTES_RETRIES)
+        .end(); // of loopDoWhile
+
+    from(ADD_NOTES_RETRIES)
+        .doTry()
+        .log("microservice request: ${body}")
+        .process(requestBgsToAddNotesProcessor())
+        .id("requestBgsToAddNotes")
+        .doCatch(BgsApiClientException.class)
+        .setBody(simple("${body.incrementTryCount()}"))
+        .log(
+            "caught: ${exception.message}: tryCount=${body.tryCount} will retry in ${body.delayMillis} ms")
+        .delay(simple("${body.delayMillis}"))
```

Comment:
Clever. Given the trouble I have had with onException and any sort of retry with that code path I rather like this way of handling it.